app/views/posts/_post.html.erb

```diff
@@ -0,0 +1,16 @@
+<p>
+  <%= post.like.all.count %></br>
+  <% if post.like.all.where(user: current_user).length===0 %>
+    <%= link_to 'Like Post', new_blog_post_like_path(post.blog, post) %>
+  <% else %>
+    <%= link_to 'Unlike post', blog_post_like_path(post.blog, post), method: :delete %>
+  <% end %>
+  </br>
+  <%= post.user.try(:email) || "deleted_user" %></br>
+  <%= post.body %></br>
+  <% if post.user==current_user || current_user==post.blog.user %>
```

Comment:
Out of curiosity, what's a scenario where `post.user` and `post.blog.user` might be different? Is the idea that people can make posts on other people's blogs?

Reply:
Yeah, that was the idea, sorry if I interpreted requirements incorrectly with this behaviour. If that was meant to be restricted I can make changes.

Reply:
Nope, that's okay. We weren't too strict on the requirements, partly to see how people would interpret them. This would be pretty unusual behaviour for any kind of traditional blog, but makes sense more as a facebook style "anyone can post on your wall including yourself" thing.